test/test_gpu3.cpp

```diff
@@ -9662,18 +9662,29 @@ TEST_F(NVFuserTest, SoftmaxDropout) {
   NVF_CHECK(reduction_params, "Reduction schedule was not generated!");
   scheduleInnerPersistentKernel(&fusion, *reduction_params);
 
-  GpuLower gpulw(&fusion);
-  auto all_exprs = KernelExprVisitor::getAllExprs(gpulw.kernel());
-  std::unordered_set<TensorView*> rng_consumers;
-  for (const auto& expr : all_exprs) {
-    if (expr->isA<RNGOp>() && expr->output(0)->isA<TensorView>()) {
-      TensorView* tv = expr->output(0)->as<TensorView>();
-      rng_consumers.insert(tv);
+  // check RNGOp is predicated without else branch.
+  class PredicateChecker : public kir::IrVisitor {
+   public:
+    using kir::IrVisitor::handle;
+    bool predicate_rngop = false;
+
+   private:
+    void handle(RNGOp* uop) final {
+      for (auto expr : scope_exprs_) {
```

Comment:
Shouldn't we only check the innermost `IfThenElse`?

Reply:
I think the key is to find the one in the `else` branch. So here I used `expr->as<kir::IfThenElse>()->hasElse()`. It captures error if predicate is eliminated.